server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1364,5 +1364,12 @@ public void calculateExportedForOwnersPools(Owner owner) {
         q.executeUpdate();
     }
 
-
+    public void markCertificatesDirtyForOwnerProductId(Owner owner, String productId) {
+        String statement = "update Entitlement e set e.dirty=true where e.pool.id in " +
+            "(select p.id from Pool p where p.product.id in (:productId)) and e.owner = :owner";
```

Comment:
Like the other branch, we're not testing against derived products. Also, if we only have one product ID, we don't need to use IN here on the inner query.